src/app/modules/lti/pages/lti/lti.component.ts

```diff
@@ -0,0 +1,105 @@
+import { GetResultsService, Result } from 'src/app/modules/item/http-services/get-results.service';
+import { bestAttemptFromResults, implicitResultStart } from 'src/app/shared/helpers/attempts';
+import { ensureDefined } from 'src/app/shared/helpers/null-undefined-predicates';
+import { errorState } from 'src/app/shared/helpers/state';
+import { ResultActionsService } from 'src/app/shared/http-services/result-actions.service';
+import { mapToFetchState } from 'src/app/shared/operators/state';
+import { fullItemRoute } from 'src/app/shared/routing/item-route';
+import { ItemRouter } from 'src/app/shared/routing/item-router';
+import { LayoutService } from 'src/app/shared/services/layout.service';
+
+const explicitEntryWithNoResultError = new Error('no result');
+const noChildError = new Error('no first child');
+const noContentIdError = new Error('no content id');
+type LTIError = 'no content id' | 'fetch error' | 'explicit entry with no result' | 'no child';
+
+@Component({
+  selector: 'alg-lti',
+  templateUrl: './lti.component.html',
+  styleUrls: [ './lti.component.scss' ],
+})
+export class LTIComponent {
+
+  readonly navigationData$ = this.activatedRoute.queryParamMap.pipe(
+    switchMap(queryParams => {
+      const contentId = queryParams.get('content_id');
+      if (!contentId) return of(errorState(noContentIdError));
+      return this.getNavigationData(contentId).pipe(mapToFetchState());
+    }),
+  );
+  readonly error$ = this.navigationData$.pipe(
+    map((state): LTIError | undefined => {
+      if (!state.isError) return undefined;
+      switch (state.error) {
+        case noContentIdError: return 'no content id';
+        case noChildError: return 'no child';
+        case explicitEntryWithNoResultError: return 'explicit entry with no result';
+        default: return 'fetch error';
+      }
+    })
+  );
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private itemRouter: ItemRouter,
+    private layoutService: LayoutService,
+    private getItemByIdService: GetItemByIdService,
+    private getResultsService: GetResultsService,
+    private getItemPathService: GetItemPathService,
+    private resultActionsService: ResultActionsService,
+    private getItemChildrenService: GetItemChildrenService,
+  ) {
+    this.layoutService.toggleTopRightControls(false);
+    this.layoutService.toggleFullFrameContent(true, false);
+
+    this.navigationData$.subscribe({
+      next: state => {
+        if (!state.isReady) return;
+        const { firstChild, path, result } = state.data;
+        const itemRoute = fullItemRoute('activity', firstChild.id, [ ...path, state.data.itemId ], { attemptId: result.attemptId });
+        this.itemRouter.navigateTo(itemRoute, { navExtras: { replaceUrl: true } });
+      },
+    });
+  }
+
+  private getNavigationData(itemId: string): Observable<{ itemId: string, firstChild: ItemChild, path: string[], result: Result }> {
```

Comment:
it seems to do all the call 3x 
![Screenshot 2021-12-15 at 18 15 54](https://user-images.githubusercontent.com/1053150/146233468-c5605c68-3522-40b8-9291-842aeb8bde5a.png)

Reply:
Fixed